remote.go

```diff
@@ -117,6 +117,9 @@ type FetchOptions struct {
 
 	// Headers are extra headers for the fetch operation.
 	Headers []string
+
+	// Proxy options to use for this fetch operation
+	*ProxyOptions
```

Comment:
it may be better to avoid [embedding](https://golang.org/doc/effective_go.html#embedding) `ProxyOptions`, and also use a value instead of a pointer type since the zero value for `ProxyOptions` is well-defined and matches the semantics of not specifying it (also makes it consistent with all the other fields of `FetchOptions`):

```suggestion
	// ProxyOptions are the proxy options to use for this fetch operation.
	ProxyOptions ProxyOptions
```